runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -1158,6 +1158,29 @@ local loclist_type_map = {
   [DiagnosticSeverity.Hint] = 'I',
 }
 
+
+--- Clear diagnotics and diagnostic cache
+---
+--- Handles saving diagnostics from multiple clients in the same buffer.
+---@param client_id number
+---@param buffer_client_map table map of buffers to active clients
+function M.stop(client_id, buffer_client_map)
+  local active_buffers = {}
+  for bufnr, client_ids in pairs(buffer_client_map) do
+    if client_ids[client_id] then
+      table.insert(active_buffers, bufnr)
+    end
+    client_ids[client_id] = nil
+  end
```

Comment:
can you put this all in the `schedule`d function?

Otherwise you can have weird inconsistent state possible, which seems confusing.

Reply:
Done

Reply:
I also simplified the function a bit.